package/gluon-offline-ssid/luasrc/usr/bin/gluon-offline-ssid

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/lua
+
+local uci = require("simple-uci").cursor()
+local util = require 'gluon.util'
+
+function safety_exit(t)
+  io.write(t .. ", exiting with error code 2")
+  os.exit(2)
+end
+
+-- return_value is optional to return instead of the output
+function shell_exec(command, return_value)
+  if return_value then
+    return os.execute(command .. ' &> /dev/null')
+  end 
+  local file = assert(io.popen(command, 'r'))
+  local output = file:read('*all')
+  file:close()
+  return output
```

Comment:
doesn't the file have to be closed before the function returns?

Reply:
If you don't close a file descriptor in Lua, it will either be closed by the garbage collector if you don't have a reference anymore or it will be closed by a POSIX kernel on program termination.
I assumed the garbage collector would be more clever with Lua 5.1 and directly close the file as it is not associated with a variable, but I did some testing and found out that the garbage collector still only runs "from time to time". Thus you can leave the code as is, but it wouldn't really hurt having a few open file descriptors.

Reply:
ok, then we leave it as it is